src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -825,12 +853,16 @@ class module_sram : public module_impl
   std::shared_ptr<module_impl> m_parent;
   xrt::hw_context m_hwctx;
 
+  static constexpr size_t m_scratchmem_size = 512 * 1024;
```

Comment:
we should get the size from elf instead of hard code it in the case we want to save different size of l2 memory

Reply:
Elf currently does not have the size information. This should be done in a follow up PR.